### ISSUE TITLE
aws_cloudwatch_log_group creation blocked by SCP

### ISSUE THREAD

**Issue by lanejlanej** [hashicorp/terraform-provider-aws#19109]
In the environment I am working in an AWS SCP exists to prevent creation of a cloudwatch log group if it is not associated with a set of desired tags, at creation time. With this SCP in place if an attempt is made to create a cloudwatch log group using the terraform aws_cloudwatch_log_group, with the desired tags specified, it is blocked. This suggests that behind the scenes the log group is created, and then tagged in a secondary action. If this is the case can it be remediated such that the tags are attached at creation time.

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform CLI and Terraform AWS Provider Version

Terraform version: 0.13.5
Provider version: hashicorp/aws v2.70.0

### Affected Resource(s)

* aws_cloudwatch_log_group

### Terraform Configuration Files

resource "aws_cloudwatch_log_group" "log_group_with_desired_tag" {
  name              = "/aws/lambda/log_group_with_desired_tag"
  retention_in_days = 30
  tags = {
      "desired_tag" = "desired_value"
    }
}

********** AWS SCP ***************
      {
         "Sid":"DenyNoDesiredTag",
         "Effect":"Deny",
         "Action":[
            "logs:CreateLogGroup"
         ],
         "Resource": "*",
         "Condition":{
            "Null":{
               "aws:RequestTag/desired_tag": "true"
            }
         }
      }

### Expected Behavior

Resource creation should not have been blocked by the SCP

### Actual Behavior

Resource creation was blocked by SCP

### Steps to Reproduce

Create an AWS SCP similar to above.
Attempt to deploy an aws_cloudwatch_log_group using code as above.

1. `terraform apply`

**Comment by gdavison** [comment]
Hi @lanejlanej, it looks like you're using version 2.70.0 of the provider, which was released 10 July 2020. We are currently at version 3.37.0, and it does set the tags at creation time. Can you please try applying your configuration with the current version?

**Comment by lanejlanej** [comment]
Hi Graham,

Thanks for the swift response.  The terraform is running in a pipeline and
I cannot change the version directly at the moment. However, I will test it
out as soon as I can.

**Comment by lanejlanej** [comment]
I can confirm that the log group creation is successful with newer provider (3.38.0) when the SCP is applied. So, closing this.

**Comment by lanejlanej** [comment]
Sorry, actually turns out there was a problem with the previous test and the SCP wasn't attached. With the SCP attached the creation of cloudwatch log groups is still blocked by the SCP, although the mandatory tags are specified in the terraform code.

**Comment by gdavison** [comment]
Hi @lanejlanej, just getting back to this. Looking at your SCP, I think it might to be saying `Deny` if `aws:RequestTag/desired_tag` is present.

**Comment by lanejlanej** [comment]
Hi @gdavison, thanks for quick response. Pretty sure that the syntax of the SCP statement and condition is correct (we use similar elsewhere against ec2:RunInstances eg.), but agree it is confusing. What it basically says in the condition is 'if it is the case that the `aws:RequestTag/desired_tag` is null, ie. doesn't exist, then `Deny`'.

**Comment by gdavison** [comment]
Since we haven't received a response, we're going to assume that this has been resolved. If you are still experiencing this problem, please open a new issue.
